nexus/codec.go

```diff
@@ -0,0 +1,204 @@
+package nexus
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"reflect"
+)
+
+// A Stream is a container for a header and a reader.
+// It is used to stream inputs and outputs in the various client and server APIs.
+type Stream struct {
```

Comment:
Not the biggest fan of "stream" here as it implies long-lived endpoints w/ messages sent at differing times. Would prefer `Body` or `Content` or something.

Reply:
Yeah, I'm changing to `Content`.